File: dbflow-rx/src/main/java/com/raizlabs/android/dbflow/rx/language/RXSQLite.java
package com.raizlabs.android.dbflow.rx.language;

import androidx.annotation.NonNull;

import com.raizlabs.android.dbflow.sql.queriable.ModelQueriable;
import com.raizlabs.android.dbflow.sql.queriable.Queriable;

/**
 * Description: Main entry point to wrap our queries in RX.
 */
public class RXSQLite {

    @NonNull
    public static <T> RXModelQueriableImpl<T> rx(ModelQueriable<T> modelQueriable) {
        return new RXModelQueriableImpl<>(modelQueriable);
    }

    @NonNull
    public static RXQueriableImpl rx(Class<?> table, Queriable queriable) {
        return new RXQueriableImpl(table, queriable);
    }
}

File: dbflow-rx/src/main/java/com/raizlabs/android/dbflow/rx/language/RXSQLite.kt
@file:JvmName("RXSQLite")

package com.raizlabs.android.dbflow.rx.language

import com.raizlabs.android.dbflow.query.BaseQueriable
import com.raizlabs.android.dbflow.query.ModelQueriable
import com.raizlabs.android.dbflow.query.Queriable

fun <T : Any> ModelQueriable<T>.rx(): RXModelQueriableImpl<T> = RXModelQueriableImpl<T>(this)

fun Queriable.rx(): RXQueriableImpl = RXQueriableImpl(this)

fun <T : Any> BaseQueriable<T>.rxBaseQueriable() = rx()
